### DISCUSSION TITLE
Replace color tool (Eraser + modifier or brand new tool)

### DISCUSSION THREAD

**Discussion by hsandt** [Orama-Interactive/Pixelorama#993] [Ideas]
On Aseprite, you can select the Eraser tool and paint with right-click to replace Primary with Secondary color (a behavior inherited from the old MS Paint). This is very convenient in order to change existing colors locally or simulate switching the whole color palette (using a giant Eraser brush).

In Pixelorama, right-click is same as left-click, just uses right tool, so there is not that subtlety. Maybe we could have another modifier (such as holding Ctrl or Super - avoid Alt which drags window on several Linux desktops by default), or even better a dedicated tool Replace Color (which would need an extra "color to replace" option OR assume the *other* side's color is the one to replace).

For full color replacement, we would need something like Indexed color mode: https://github.com/Orama-Interactive/Pixelorama/discussions/572

**Comment by Variable-ind** [comment]
Hy 👋, have you tried the "swappy" extension?
It's available in [VariableStore](https://variable-interactive.itch.io/pixelorama-extensions) (it's a place where you can download pixelorama extensions in one place)


https://github.com/Orama-Interactive/Pixelorama/assets/77773850/7f896235-aaa4-41ca-a7ea-3dbcabc6c4d6

**Comment by OverloadedOrama** [comment]
This is now possible by using the pencil as the right tool and enabling alpha lock from the global tool options. It is possible to add a keyboard shortcut to alpha lock from the Preferences, but we could add a default one as well. You could also use the Color Replace mode of the Shading tool, if you are working with colors in a palette. As for indexed mode, it has been implemented for version 1.1.
